src/blockrelay/blockrelay_common.cpp

```diff
@@ -104,26 +106,21 @@ bool ThinTypeRelay::HasBlockRelayTimerExpired(const uint256 &hash)
 
 bool ThinTypeRelay::IsBlockRelayTimerEnabled()
 {
-    // Only engage the timer if at least one thin type relay is active.
-    if (!IsThinBlocksEnabled() && !IsGrapheneBlockEnabled())
+    // Only engage the timer if one or more, but not all, thin type relays are active.
+    // If all types are active, or all inactive, then we do not need the timer.
+    // Generally speaking all types will be active and we can return early.
+    if (IsThinBlocksEnabled() && IsGrapheneBlockEnabled() && IsCompactBlocksEnabled())
         return false;
```

Comment:
ok, fixed the log entries so they don't get printed out if the timer is disabled.

> @ptschip
> 
> > It's just the log statement getting printed out, but I can fix that.
> 
> ok thanks. But I think that `2019-01-31 14:24:00 Starting Preferential Block Relay timer (944 millis)` it is only be produce if `HasBlockRelayTimerExpired()` is executed and I think we could avoid to execute the entire function in case of the timer is disabled.
> 
> > As for @bissias  PR, I think he'll have to update and rebase after this  one gets merged.
> 
> Agreed.